src/Mado/QueryBundle/Services/IdsChecker.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace Mado\QueryBundle\Services;
+
+use Mado\QueryBundle\Objects\Filter;
+use Mado\QueryBundle\Exceptions\ForbiddenContentException;
+
+class IdsChecker
+{
+    private $idsMustBeSubset = true;
+
+    private $filtering;
+
+    private $additionalFilter;
+
+    private $filterKey;
+
+    private $finalFilterIds;
+
+    public function __construct()
+    {
+        $this->idsMustBeSubset = true;
+    }
+
+    public function setFiltering($filtering)
+    {
+        $this->filtering = $filtering;
+    }
+
+    public function setObjectFilter(Filter $additionalFilter)
+    {
+        $this->additionalFilter = $additionalFilter;
+    }
+
+    public function setFilterKey($filterKey)
+    {
+        $this->filterKey = $filterKey;
+    }
+
+    public function validateIds()
+    {
+        $rawFilteredIds = $this->additionalFilter->getIds();
+
+        foreach ($this->filtering as $key => $queryStringIds) {
+            $querystringIds = explode(',', $queryStringIds);
+            $additionalFiltersIds = explode(',', $this->additionalFilter->getIds());
+            foreach ($querystringIds as $requestedId) {
+                if ($this->additionalFilter->getOperator() == 'list') {
+                    if (!in_array($requestedId, $additionalFiltersIds)) {
```

Comment:
is possible to move some piece of code into other functions or objects?

Reply:
This class is already the result of a refactoring. I am afraid that another decoupling, now, could be an over engineered solution. Yes, it is possibile. Not so useful now.